modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java

```diff
@@ -164,10 +164,7 @@ public int doStartTag() throws JspException {
 				_stockQuantity = stockQuantity.intValue();
 
 				if (!_disabled) {
-					_disabled =
-						(!_productSettingsModel.isBackOrders() &&
-						 (_stockQuantity <= 0)) ||
-						!cpSku.isPublished() || !cpSku.isPurchasable();
+					_disabled = !cpSku.isPublished();
```

Comment:
let's get rid of the _disabled and add the _published property

Reply:
I do not think we should get rid of it. With my changes in the PR, the button is disabled when it fails permission checks or is not published.